mempool/mempool.go

```diff
@@ -327,9 +326,11 @@ func (mem *Mempool) notifyTxsAvailable() {
 	if mem.Size() == 0 {
 		panic("notified txs available but mempool is empty!")
 	}
-	if mem.txsAvailable != nil && !mem.notifiedTxsAvailable {
-		mem.notifiedTxsAvailable = true
-		mem.txsAvailable <- mem.height + 1
+	if mem.txsAvailable != nil {
+		select {
+		case mem.txsAvailable <- mem.height + 1:
+		default:
+		}
```

Comment:
This non-blocking send might break our guarantee on the consumer side, which if I'm not mistaken is only the consensusState. What happens if the select on the other side is never ready when the new height is send, will that break our consensus code?

Reply:
the channel is buffered with capacity 1 so the first send should go through no matter what.

really we don't want a second send to even happen, which is what the boolean was supposed to prevent before, but on closer inspection is didn't seem quite thread safe, though I couldn't generate a race condition.

Tbh I'm not 100% this is the right solution and I agree its a bit troubling. How can we make it this dependency more explicit as you suggest ?